drivers/SmartThings/zigbee-presence-sensor/src/aqara/init.lua

```diff
@@ -0,0 +1,132 @@
+local RESET_PRESENCE = "stse.resetPresence"
+local APP_DISTANCE = "stse.approachDistance"
+
+local FINGERPRINTS = {
+  { mfr = "aqara", model = "lumi.motion.ac01" }
+}
+
+local is_aqara_products = function(opts, driver, device, ...)
+  for _, fingerprint in ipairs(FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+local function device_init(driver, device)
+  -- no action
+end
+
+local function device_added(driver, device)
+  -- private protocol enable
+  device:send(cluster_base.write_manufacturer_specific_attribute(device,
+    PRIVATE_CLUSTER_ID, PRIVATE_ATTRIBUTE_ID, MFG_CODE, data_types.Uint8, 1))
+  -- init
+  device:send(cluster_base.write_manufacturer_specific_attribute(device,
+    PRIVATE_CLUSTER_ID, MONITORING_MODE, MFG_CODE, data_types.Uint8, 0))
+  device:send(cluster_base.write_manufacturer_specific_attribute(device,
+    PRIVATE_CLUSTER_ID, RESET_MODE, MFG_CODE, data_types.Uint8, 1))
+end
+
+local function device_info_changed(driver, device, event, args)
+  if device.preferences ~= nil then
+    if device.preferences[SENSITIVITY] ~= args.old_st_store.preferences[SENSITIVITY] then
+      device:send(cluster_base.write_manufacturer_specific_attribute(device,
+        PRIVATE_CLUSTER_ID, 0x010C, MFG_CODE, data_types.Uint8, tonumber(device.preferences[SENSITIVITY])))
+    end
+    if device.preferences[RESET_PRESENCE] ~= args.old_st_store.preferences[RESET_PRESENCE] then
+      device:send(cluster_base.write_manufacturer_specific_attribute(device,
+        PRIVATE_CLUSTER_ID, RESET_MODE, MFG_CODE, data_types.Uint8, 0x01))
+    end
+    if device.preferences[APP_DISTANCE] ~= args.old_st_store.preferences[APP_DISTANCE] then
+      device:send(cluster_base.write_manufacturer_specific_attribute(device,
+        PRIVATE_CLUSTER_ID, 0x0146, MFG_CODE, data_types.Uint8, tonumber(device.preferences[APP_DISTANCE])))
+    end
+  end
+end
+
+local function presence_monitor_handler(driver, device, value, zb_rx)
+  local val = value.value
+
+  if val == 0 then
+    device:emit_event(PresenceSensor.presence("not present"))
+  elseif val == 1 then
+    device:emit_event(PresenceSensor.presence("present"))
+  end
+end
+
+local function move_monitor_handler(driver, device, value, zb_rx)
+  local val = value.value
+
+  if val < 2 or val == 6 or val == 7 then
+    if val == 0 then
+      device:emit_event(MovementSensor.movement("enter"))
+    elseif val == 1 then
+      device:emit_event(MovementSensor.movement("leave"))
+    elseif val == 6 then
+      device:emit_event(MovementSensor.movement("approaching"))
+    elseif val == 7 then
+      device:emit_event(MovementSensor.movement("goingAway"))
+    end
+    local movement_timer = device:get_field(MOVEMENT_TIMER)
+    if movement_timer then
+      device.thread:cancel_timer(movement_timer)
+      device:set_field(MOVEMENT_TIMER, nil, { persist = true })
```

Comment:
this is about to be overwritten anyway, so you don't need to do it here

Reply:
Okay. I'll fix it.